Update README.md with Bulk Export instructions (#48)
inferno-framework/davinci-pdex-test-kit: README.md

opening the SMART App Launch STU2.2 suite in another tab and running Test Group 1 Standalone Launch. Use
the FHIR URL and SMART Client Id inputs from the Payer Client test's dialog box. Grab
the bearer token from test 1.2.06 output `standalone_access_token`. 
4. On the Payer Client suite, follow the user action required to indicate you have completed client connection.
5. Now on the Payer Server suite, select the "PDex Payer Server Preset for Client Tests" preset, start
Test Group 1.1 $member-match, and enter the access token you grabbed from step 3. Submit the inputs to run 1.1
6. After the Client suite receives the $member-match request, which you can confirm on the Server suite's side,
you can run any of the other Test Groups to simulate that portion of the PDex IG:
  * 1.2 for a single clinical FHIR query on an Encounter resource
  * 1.3 for the Patient `$everything` operation
  * 1.4 for the `$export` operation from bulk data export.
  * 2.2 for the US Core and PDex `GET` API.
7. Once you have executed any of the above tests, follow the User Action Required dialog box on the Client suite
to indicate you have finished making requests.

> Troubleshooting: if you get the error `Unable to find test run with identifier...` it is due to known race condition bug

Reviewer: There are lots of reasons that this particular error could come up, so I think being a bit softer is a good idea.

```suggestion
Troubleshooting: if you get the error `Unable to find test run with identifier...` it may be due to known race condition bug
```

   (NOTE: this decision can be made independently from whether to run the test kit with 
   docker or using Ruby).

If you are running the Server Bulk Export tests, you **must** use a local Inferno Reference Server
with **read-write mode enabled** by adding `READ_ONLY=false` to the environment and with
the **timeout increased to 600 seconds**.

## Running Server Tests and Client Tests against each other

1. Open the Payer Server and Payer Client test suites in separate windows or tabs.
2. Select "Run All Tests" to start the Payer Client suite.
3. Use the SMART App Launch Test Kit to perform the authentication handshake. You can do this by running
opening the SMART App Launch STU2.2 suite in another tab and running Test Group 1 Standalone Launch. Use
the FHIR URL and SMART Client Id inputs from the Payer Client test's dialog box. Grab

> the bearer token from test 1.2.06 output `standalonne_access_token`. 

Reviewer: ```suggestion
the bearer token from test 1.2.06 output `standalone_access_token`. 
```

4. On the Payer Client suite, follow the user action required to indicate you have completed client connection.
5. Now on the Payer Server suite, select the "PDex Payer Server Preset for Client Tests" preset, start
Test Group 1.1 $member-match, and enter the access token you grabbed from step 3. Submit the inputs to run 1.1
6. After the Client suite receives the $member-match request, which you can confirm on the Server suite's side,
you can run any of the other Test Groups to simulate that portion of the PDex IG:
  * 1.2 for a single clinical FHIR query on an Encounter resource
  * 1.3 for the Patient `$everything` operation
  * 1.4 for the `$export` operation from bulk data export.
  * 2.2 for the US Core and PDex `GET` API.
7. Once you have executed any of the above tests, follow the User Action Required dialog box on the Client suite
to indicate you have finished making requests.

Since this exercise requires running two test kits which causes port conflicts by default, you are advised to do
it on [the ONC website](https://inferno.healthit.gov/test-kits/) or locally on the

> [inferno platform](https://github.com/inferno-framework/inferno-platform-template/).

Reviewer: But it is all the PDex test kit, so you don't have to run more than 1 test kit at a time locally to do this. I think you can remove this paragraph unless I'm missing something.

Author: That is a good point - removed

some server test suite requests. As a work-around, run the server test suite a second time while the
client is waiting for clinical data requests during test 1.2.

> ## Running the Bulk Export Server Tests

Reviewer: ```suggestion
## Server Bulk Export Test Demonstration
```

> ## Running the Bulk Export Server Tests

Reviewer: Another demonstration that would be valuable would be to show how to run against the PDex Client tests. I tried this and it didn't get very far because the bulk tests stop the workflow if they detect even small things like bad HTTP statuses (see [FI-3949](https://oncprojectracking.healthit.gov/support/browse/FI-3949)). Probably doesn't make sense to resolve that or fixes to make this test kit return the right statuses as a part of this ticket, but wanted to mention it.

> These tests require read-write access to FHIR server with `Patient/$export` enabled, or with permissions to those actions assigned to a Bulk Data Authorization Access Token. To view an example run of these tests:

Reviewer: Why do they require write access? Is that a HAPI requirement around bulk export? I think all of this is in the context of our reference server, so I would make that clear:
```suggestion
The Server Bulk Export tests (group 1.4) can be run against the Inferno Reference Server as long as it is running with write enabled:
```

Author: You're right this is only for our reference server not from the FHIR spec
